src/zkevm_specs/evm/execution/end_block.py

```diff
@@ -1,12 +1,24 @@
-from ...util import FQ
+from ...util import FQ, N_BYTES_GAS
 from ..instruction import Instruction, Transition
-from ..table import CallContextFieldTag, TxTableRow, TxContextFieldTag
+from ..table import (
+    CallContextFieldTag,
+    TxTableRow,
+    TxContextFieldTag,
+    BlockContextFieldTag,
+    TxReceiptFieldTag,
+)
 from typing import Set
 
 # EndBlock is an execution state that constraints the following:
-# A. Once the EndBlock state is reached, there's no other execution states appearing until the end of the EVM Circuit.  In particular, after the first EndBlock, there will be no new lookups to the rw_table.
-# B. The number of meaningful entries (non-padding) in the rw_table match the rw_counter after the EndBlock state is processed.
-# C. The number of txs processed by the EVM Circuit match the number of txs in the TxTable
+# A. Once the EndBlock state is reached, there's no other execution states
+# appearing until the end of the EVM Circuit.  In particular, after the first
+# EndBlock, there will be no new lookups to the rw_table.
+#
+# B. The number of meaningful entries (non-padding) in the rw_table match the
+# rw_counter after the EndBlock state is processed.
+#
+# C. The number of txs processed by the EVM Circuit match the number of txs in
+# the TxTable
```

Comment:
Since this comment serves as spec/doc for this execution state, could you add a point `D` mentioning the gas check?